Added support for Crystal 0.36.1 - 1.x.x

crimson-knight opened the pull request:
# What does this PR do?
I updated the Crystal version so that I can use this shard in my new Amber application that uses Crystal 1.0

# Any background context you want to provide?
Updated the shard file, and was able to setup a local test DB and run all specs with the following results:
```bash
Finished in 7.83 seconds
1789 examples, 0 failures, 0 errors, 94 pending
```

# Release notes
Adds support for any official releases of Crystal from 0.36.1 - 1.x.x specifically allowing for any of the minor releases because Crystal 1.x is supposed to avoid any breaking changes until Crystal 2.0

imdrasil commented:
Good catch :+1: but let's make restriction less strict - `crystal: ">= 0.36.1"`

crimson-knight commented:
I've updated the restriction to match the lowest support crystal version in the readme file and removed the 2.x cap on the crystal version. All tests are passing as before.
